rfq/oracle.go

```diff
@@ -232,13 +220,28 @@ func (r *RpcPriceOracle) QueryAskPrice(ctx context.Context,
 
 	// Construct the RPC rate tick hint.
 	var rateTickHint *oraclerpc.RateTick
-	if bidPrice != nil {
+	if assetRateHint != nil {
 		// Compute an expiry time using the default expiry delay.
 		expiryTimestamp := uint64(time.Now().Unix()) +
 			defaultRateTickExpirySeconds
 
+		// Marshal the subject asset rate.
+		subjectAssetRate, err := oraclerpc.MarshalBigIntFixedPoint(
+			*assetRateHint,
+		)
+		if err != nil {
+			return nil, err
+		}
+
 		rateTickHint = &oraclerpc.RateTick{
-			Rate:            uint64(*bidPrice),
+			SubjectAssetRate: subjectAssetRate,
+
+			// For now, we only support BTC as the payment asset.
+			PaymentAssetRate: &oraclerpc.FixedPoint{
+				Coefficient: rfqmsg.MilliSatPerBtc.Uint64(),
+				Scale:       0,
```

Comment:
Assuming a scale factor again.

Reply:
`rfqmsg.MilliSatPerBtc` is of type big int fixed-point in the latest commits. It's now defined as:
```
MilliSatPerBtc = rfqmath.FixedPointFromUint64[rfqmath.BigInt](100, 9)
```
does that address your concern?